src/infrastructure/performance/memory-monitor.ts

```
@@ -91,24 +91,23 @@ const makeMemoryMonitorService = Effect.gen(function* () {
-			//TODO: can we do better here??
-			// return Effect.fail(new GarbageCollectionFailureError("unknown", memoryBefore)) // Unreachable - commented out
-		}
+		// Use Effect.try for safe garbage collection with proper error handling
+		return Effect.try({
+			try: () => {
+				if (typeof global !== "undefined" && global.gc) {
+					global.gc()
+					const memoryAfter = getCurrentMemoryUsage().heapUsed
+					return {memoryBefore: createByteAmount(memoryBefore), memoryAfter: createByteAmount(memoryAfter)}
+				} else {
+					throw new GarbageCollectionNotAvailableError(createByteAmount(memoryBefore))
```

ESLint

[error] 🚨 BANNED: throw statements. Use Effect.fail() or Effect.die() instead for proper error handling in Effect.TS pipelines.

(no-restricted-syntax)